fractions/javaee/undertow/src/main/java/org/wildfly/swarm/undertow/runtime/HttpSecurityPreparer.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2015-2017 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.undertow.runtime;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.Node;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.wildfly.swarm.spi.api.ArchivePreparer;
+import org.wildfly.swarm.spi.api.JARArchive;
+import org.wildfly.swarm.spi.api.annotations.Configurable;
+import org.wildfly.swarm.undertow.descriptors.JBossWebAsset;
+import org.wildfly.swarm.undertow.descriptors.SecurityConstraint;
+import org.wildfly.swarm.undertow.descriptors.WebXmlAsset;
+
+/**
+ * @author Heiko Braun
+ */
+@ApplicationScoped
+public class HttpSecurityPreparer implements ArchivePreparer {
+
+    private static final Logger LOG = Logger.getLogger(HttpSecurityPreparer.class);
+
+    private final String[] SUPPORTED_AUTH_METHODS = new String[] {"BASIC", "DIGEST", "FORM", "KEYCLOAK"};
+
+    @Override
+    public void prepareArchive(Archive<?> archive) {
+
+        if (httpConfig == null || httpConfig.isEmpty()) {
+            return;
+        }
+
+        if (!httpConfig.containsKey("login-config")) {
+            throw new RuntimeException("Syntax error: HTTP security constraints requires a login-config declaration");
+        }
+
+        // unsupported auth method
+        Map<String, Object> loginConfig = (Map<String, Object>)httpConfig.get("login-config");
+        String authMethod = (String)loginConfig.getOrDefault("auth-method", "NONE");
+        boolean isSupported = false;
+        for (String supported : SUPPORTED_AUTH_METHODS) {
+            if (authMethod.equals(supported)) {
+                isSupported = true;
+                break;
+            }
+        }
+
+        if (!isSupported) {
+            LOG.warn("Ignoring unsupported auth-method: " + authMethod);
+            return;
+        }
+
+        WebXmlAsset webXml = findWebXml(archive);
+        JBossWebAsset jbossWeb = findJBossWeb(archive);
```

Comment:
These shouldn't be needed.

It should be possible to:

  ```
WARArchive war = archive.as(WARArchive.war);
  war.addContextParam("resteasy.scan", "true");
```

Anything in web.xml or jboss-web.xml should be accessible from the WARArchive, making the find methods redundant.